nextcord/utils.py

```diff
@@ -384,6 +384,21 @@ def find(predicate: Callable[[T], Any], seq: Iterable[T]) -> Optional[T]:
     return None
 
 
+def _key_fmt(key: str) -> str:
+    # Private helper for `nextcord.utils.get`. Formats the attribute
+    # names in a content aware manner. When special variables are
+    # provided, double trailing and leading underscores are ignored.
+    # ex. "__class______name__" -> "__class__.__name__"
+    # In case of underscore conflicts, it will assume trailing underscores.
+    # ex. "_privateattr___subattr" -> "_privateattr_.subattr"
+
+    if key.startswith("__") and key.endswith("__"):
+        return re.sub(r"((__){1}__(__){1})", "__.__", key)
+    else:
+        slices = re.split(r"(?=__)", key)
+        return "".join([s.replace("__", ".") for s in slices])
```

Comment:
I may be wrong, but can't this be simplified using a single regex.

It seems like you are trying to make it so that a `__` that's not followed by an underscore will be replaced with a `.`, so:

```py
re.sub(r"__(?!_)", ".", key)
```

This will make `trailing___attr` output `trailing_.attr`.

Reply:
I'll definitely admit that I only have a very basic understanding of regex, so there definitely is room to simply my expressions. You're right that I used a greedy lookahead to match `__` not followed by another underscore.